lib/catalog.js

```diff
@@ -0,0 +1,68 @@
+const { fetch } = require('./fetch');
+
+const { httpBody } = require('./response-transforms');
+
+const status = conn => {
+  const headers = conn.headers();
+  headers.set('Accept', 'application/json');
+
+  return fetch(conn.request('admin', 'catalog', 'status'), {
+    headers,
+  }).then(httpBody);
+};
+
+const reload = (conn, provider) => {
+  const headers = conn.headers();
+  headers.set('Content-Type', 'application/json');
+
+  return fetch(conn.request('admin', 'catalog', 'reload'), {
+    method: 'POST',
+    body: JSON.stringify({ provider }),
+    headers,
+  }).then(httpBody);
+};
+
+const addCredentials = (conn, user, label) => {
```

Comment:
Copied the spelling verbatim from [the platform code](https://github.com/stardog-union/stardog/blob/79a96e23dcfca24d145edeb96b6352249be128d4/catalog/protocols/http/client/main/src/com/complexible/stardog/catalog/protocols/http/client/HttpCatalogAdminConnectionImpl.java#L116) for now... but I kinda feel like this should be `addCredential` to be consistent with `removeCredential`? The endpoint itself is just `credentials`

Reply:
i wouldn't mind if we named it `addCredential` here